roles/lib_openshift/src/class/oc_project.py

```diff
@@ -0,0 +1,183 @@
+        self.config = config
+        self._project = None
+
+    @property
+    def project(self):
+        ''' property for project'''
+        if not self._project:
+            self.get()
+        return self._project
+
+    @project.setter
+    def project(self, data):
+        ''' setter function for project propeorty'''
+        self._project = data
+
+    def exists(self):
+        ''' return whether a project exists '''
+        if self.project:
+            return True
+
+        return False
+
+    def get(self):
+        '''return project '''
+        result = self._get(self.kind, self.config.name)
+
+        if result['returncode'] == 0:
+            self.project = Project(content=result['results'][0])
+            result['results'] = self.project.yaml_dict
+
+        elif 'namespaces "%s" not found' % self.config.name in result['stderr']:
+            result = {'results': [], 'returncode': 0}
+
+        return result
+
+    def delete(self):
+        '''delete the object'''
+        return self._delete(self.kind, self.config.name)
+
+    def create(self):
+        '''create a project '''
+        cmd = ['new-project', self.config.name]
+        cmd.extend(self.config.to_option_list())
+
+        return self.openshift_cmd(cmd, oadm=True)
+
+    def update(self):
+        '''update a project '''
+
+        self.project.update_annotation('display-name', self.config.config_options['display_name']['value'])
+        self.project.update_annotation('description', self.config.config_options['description']['value'])
+
+        # work around for immutable project field
+        if self.config.config_options['node_selector']['value']:
+            self.project.update_annotation('node-selector', self.config.config_options['node_selector']['value'])
+        else:
+            self.project.update_annotation('node-selector', self.project.find_annotation('node-selector'))
+
+        return self._replace_content(self.kind, self.config.namespace, self.project.yaml_dict)
+
+    def needs_update(self):
+        ''' verify an update is needed '''
+        result = self.project.find_annotation("display-name")
+        if result != self.config.config_options['display_name']['value']:
+            return True
+
+        result = self.project.find_annotation("description")
+        if result != self.config.config_options['description']['value']:
+            return True
+
+        result = self.project.find_annotation("node-selector")
+        if result != self.config.config_options['node_selector']['value']:
+            return True
+
+        # Check rolebindings and policybindings
+        return False
+
+    # pylint: disable=too-many-return-statements,too-many-branches
+    @staticmethod
+    def run_ansible(params, check_mode):
+        '''run the idempotent ansible code'''
+
+        _ns = None
+        if params['node_selector'] is not None:
+            _ns = ','.join(params['node_selector'])
+
+        pconfig = ProjectConfig(params['name'],
```

Comment:
It does seem odd to use the project name as the namespace considering projects/namespaces do not have a namespace attribute set in the object metadata. Using `None` might be more appropriate if possible.